src/Equinor.ProCoSys.Config/CommWebApp/Configuration.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Equinor.ProCoSys.Config.CommWebApp
+{
+    public static class Configuration
+    {
+        [Authorize]
+        [FunctionName("CommWebAppConfiguration")]
+        public static IActionResult Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "CommWebApp/Configuration")] HttpRequest req,
+            ILogger log)
+        {
+            log.LogInformation("Processing configuration request...");
+            var authenticationHeader = req.Headers.FirstOrDefault(x => x.Key == "Authorization");
+            if (authenticationHeader.Key == null)
+            {
+                return new UnauthorizedResult();
+            }
+            var token = authenticationHeader.Value.ToString()?.Replace("Bearer ", null) ?? string.Empty;
+
+            var discoveryEndpoint = Environment.GetEnvironmentVariable("DiscoveryEndpoint");
+            var audience = Environment.GetEnvironmentVariable("Audience");
+            var issuer = Environment.GetEnvironmentVariable("Issuer");
+            if (!TokenValidator.TryValidate(token, out JwtSecurityToken securityToken, discoveryEndpoint, audience, issuer))
+            {
+                return new UnauthorizedResult();
+            }
+
+            var currentUserEmail = securityToken.Claims.FirstOrDefault(x => x.Type == "upn")?.Value ?? string.Empty;
+            var currentUserDomain = string.Empty;
+            if (currentUserEmail.Length > 1)
+            {
+                currentUserDomain = currentUserEmail[(currentUserEmail.IndexOf('@') + 1)..];
+            }
+
+            var originHeader = req.HttpContext.Request.Headers.FirstOrDefault(x => x.Key == "Origin");
+            if (originHeader.Key == null)
+            {
+                return new BadRequestObjectResult("Invalid origin");
+            }
+            var environment = EnvironmentSelector.GetEnvironment(originHeader.Value);
+            if (string.IsNullOrWhiteSpace(environment))
+            {
+                return new BadRequestObjectResult("Invalid origin");
+            }
+
+            var configConnectionString = Environment.GetEnvironmentVariable("FrontendConfig");
```

Comment:
This needs to be setup using a new CommWebAppConfig with a new App Configuration instance

Reply:
This is now updated from "FrontendConfig" to "CommWebAppConfig". CommWebAppConfig has been added as a key in pcs-config-non-prod-func > Configuration, with the value being the endpoint to a separate app config instance for the Commissioning webapp.